src/test/java/se/liu/ida/hefquin/engine/data/utils/TestsForSolutionMappingsIterableWithFilter.java

```diff
@@ -0,0 +1,87 @@
+package se.liu.ida.hefquin.engine.data.utils;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+
+import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.data.impl.SolutionMappingUtils;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.*;
+
+/**
+ * This is a class with helper functions for testing SolutionMappingsIterableWithFilter
+ */
+public class TestsForSolutionMappingsIterableWithFilter
+{
+	final Var var1 = Var.alloc("v1");
+	final Var var2 = Var.alloc("v2");
+	final Var var3 = Var.alloc("v3");
+	final Var var4 = Var.alloc("v4");
+	final Var var5 = Var.alloc("v4");
+	final Var var6 = Var.alloc("v4");
```

Comment:
@chengsijin0817 is it correct that all these program variables are for the same query variable?

Reply:
Fixed.